lib/index.js

```diff
@@ -451,6 +458,25 @@ if (enabled) {
     enabled = false;
   }
 
+  // clean up any leftover socket paths
+  ao.fs = require('fs');
+  ao.fs.readdir(ao.notifications.socketDir, function (e, files) {
+    if (e) {
+      return;
+    }
+    const dir = ao.notifications.socketDir;
+    const fileOnlyPrefix = ao.notifications.socketPrefix;
+    files.filter(f => f.startsWith(fileOnlyPrefix))
+      .forEach(f => {
+        // if it's the currently active socket ignore it
+        if (ao.notifications.socket.endsWith(f)) {
+          return;
+        }
+        // delete the file ignoring any errors.
+        ao.fs.unlink(`${dir}${f}`, () => undefined);
+      });
```

Comment:
Can there be multiple socket paths?
If so, should this be an exit from the loop instead of a return from the function?

Reply:
it's possible that their might be more than one though unlikely in production. the cost of looping through them is low enough that i'll leave it in for now. sometimes when testing there is more than one. but you do make me wonder what might happen if multiple applications are running on one host - there could be more than one active socket path then; i'll need to test that the delete fails (because the file is in use) in that case.

Reply:
interesting, but ok. i can delete an existing socket path but existing client-server connections continue to work. that's good. so if multiple applications are running on a single machine then the worst case is that a different application's socket path is deleted before the client attaches to it. in that case oboe will default logging. it's ugly but information isn't lost. if the path is deleted after the client-server connection is established then there is no problem. that's a pretty short window for an error; it could be mitigated by checking the creation time of the socket path but i suspect that it's unlikely that the problem will be hit. if that assumption is wrong the timestamp is a fix.